src/Raven.Server/Documents/Handlers/TimeSeriesHandler.cs

```diff
@@ -1236,6 +1251,37 @@ public override TransactionOperationsMerger.IReplayableCommandDto<TransactionOpe
             {
                 throw new System.NotImplementedException();
             }
+
+            public void AddToDisposal(IDisposable disposable)
+            {
+                _toDispose.Add(disposable);
+            }
+            
+            public void AddToReturn(AllocatedMemoryData allocatedMemoryData)
+            {
+                _toReturn.Add(allocatedMemoryData);
+            }
+
+            public void Dispose()
+            {
+                if (_isDisposed)
+                    return;
+
+                _isDisposed = true;
+                
+                foreach (var disposable in _toDispose)
+                {
+                    disposable.Dispose();
```

Comment:
Can any of these throw? If that is even potentially possible then we won't call below code returning memory and context.

Maybe let's wrap it by ExceptionAggregator?

Reply:
We don't do that in others too often, probably this throws when something really bad happens with the memory anyway.

Reply:
okay